src/basicio.cpp

```diff
@@ -47,12 +48,13 @@ namespace fs = std::filesystem;
 #include <experimental/filesystem>
 namespace fs = std::experimental::filesystem;
 #endif
+#endif
 
 // *****************************************************************************
 // class member definitions
 namespace {
 /// @brief replace each substring of the subject that matches the given search string with the given replacement.
-void ReplaceStringInPlace(std::string& subject, std::string_view search, std::string_view replace) {
+[[maybe_unused]] void ReplaceStringInPlace(std::string& subject, std::string_view search, std::string_view replace) {
```

Comment:
probably makes sense to get rid of this as there's a single usage. Maybe convert to lambda.